### DISCUSSION TITLE
Error "inject() can only be used inside setup() or functional components" after ugrading to beta 40

### DISCUSSION THREAD

**Discussion by JeremieLitzler** [vuepress/core#977] [Q&A] [answered]
Hello,

I'm having an issue upgrading to beta 40 (I'm actually at beta 41 to solve the nprogress seen in beta 40).

1/ I have follow the setup to upgrade my config file as described in [the changelog](https://github.com/vuepress/vuepress-next/blob/main/CHANGELOG.md#200-beta40-2022-04-25) and I updated my custom theme as described in 
 [the docs](https://v2.vuepress.vuejs.org/reference/default-theme/extending.html).

- See [commit 1a5fdc7a1f9ccc803c5de56f0171ae4dbb98f0af](https://github.com/JeremieLitzler/journal/commit/1a5fdc7a1f9ccc803c5de56f0171ae4dbb98f0af) and [commit 99824a139377dd87ce3a1dd0eb0c4a9731f9dbf5](https://github.com/JeremieLitzler/journal/commit/99824a139377dd87ce3a1dd0eb0c4a9731f9dbf5)

3/ The command `npm run docs:build` is successful.

4/ But the rendering in the browser fails:
- First, I see this warning:
![image](https://user-images.githubusercontent.com/2410556/177478614-5d946817-802b-4d76-910f-9cf3db872394.png)

- Then this error is thrown:
![image](https://user-images.githubusercontent.com/2410556/177478855-152c85d5-25dd-4501-8e9c-533b2aebd130.png)

I think it has to do with my custom Layout not using setup properly. I try [this](https://github.com/JeremieLitzler/journal/commit/99824a139377dd87ce3a1dd0eb0c4a9731f9dbf5#diff-d71037b75ef4b5bfbfb60b5e94a9b5dbeea9bd66f98d6edb4990578e6ff26310) but it didn't solve it.

What am I missing?

Thank you very much for your help!

**Comment by meteorlxy** [comment, marked as answer]
First of all, you'd better clean the lockfile and node_modules then re-install again. I saw you have two different version in your lockfile: https://github.com/JeremieLitzler/journal/blob/7935d5a7e5f29e6696227520db75e2a2e921c669/yarn.lock#L1268-L1311

Then, if you still got stuck, you should provide a reproduce repo. A standalone commit is not helpful enough.

**Reply by JeremieLitzler** [reply]
@meteorlxy : 

I shall try to clean the lockfile and node_modules then re-install again.
Otherwise, I have a [reproduce repo](https://github.com/JeremieLitzler/journal/tree/upgradeToVuePressBeta39) from which I took the commit :)

**Reply by JeremieLitzler** [reply]
@meteorlxy : clean the lockfile and node_modules then re-install again solve the issue.

I added that trick to my toolbox.

Thanks a million!
